src/coreclr/jit/valuenum.cpp

```diff
@@ -7754,10 +7757,10 @@ void Compiler::fgValueNumberTree(GenTree* tree)
                                         rhsVNPair.SetBoth(vnStore->VNForExpr(compCurBB, lclVarTree->TypeGet()));
                                     }
 
+                                    lclDefSsaNum = GetSsaNumForLocalVarDef(lclVarTree);
```

Comment:
there was one failure on a big method before this change, the case looked like this:
```
ASG(LCL_VAR V01, 0) <- def
ASG(LCL_FLD V01+4, 1) with NotAField <- both def and use
Read LCL_FLD V01+8 with LclFld <- use
```
and SSA created such numbers:
```
ASG(LCL_VAR V01, 0) <- lclDefSsaNum 0
ASG(LCL_FLD V01+4, 1) with NotAField <- lclDefSsaNum 1, use ssa num 0
Read LCL_FLD V01+8 with LclFld <- use ssa num 1
```
and when we did `fgValueNumberTree` for `Read LCL_FLD V01+8` we expected data `ssa num 1` to be initialized but it was not because we used ssa num 0 for `ASG(LCL_FLD V01+4, 1)`.